swift/ios-shoppe-demo/Views/CheckoutTextViewCell.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  CheckoutTextViewCell.swift
+//  ios-shoppe-demo
+//
+//  Created on 4/10/20.
+//  Copyright © 2020 FullStory All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class CheckoutTableViewCell: UITableViewCell, UITextFieldDelegate {
+    @IBOutlet weak var addressDetailTextField: UITextField!
+
+    var tableViewController: CheckoutTableViewController?
+    var addressDetail: AddressDetail? {
+        didSet {
+            setup()
+        }
+    }
+
+    var cardDetail: CardDetail? {
+        didSet {
+            setup()
+        }
+    }
+
+    func setup() {
+        if addressDetail != nil {
```

Comment:
Since the if/else is shallow, we could go with a ternary:

```
addressDetailTextField.placeholder = addressDetail != nil ? addressDetail?.placeHolder :  cardDetail?.placeHolder
```